### ISSUE TITLE
Wasmtime 11.0.0 Rust 1.71.0: error[E0599]: no method named `append` found for struct `Arc<cap_std::fs::File>` in the current scope

### ISSUE THREAD

**Issue by jornfranke** [bytecodealliance/wasmtime#6755]
Hallo,

I try to upgrade the following simple wasmtime app to wasmtime 11.0.0:
https://github.com/ZuInnoTe/rust-wasm-dynamic-module-study/tree/main/wasm-app

However, during compilation (cargo build --release), I receive the following error in the wasmtime library (it does not seem to come from my app):
```
   Compiling wasmtime-wasi v11.0.0
error[E0599]: no method named `append` found for struct `Arc<cap_std::fs::File>` in the current scope
   --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wasmtime-wasi-11.0.0/src/preview2/filesystem.rs:247:22
    |
247 |         Ok(self.file.append(buf)? as i64 as u64)
    |                      ^^^^^^ method not found in `Arc<File>`

error[E0599]: no method named `append_vectored` found for struct `Arc<cap_std::fs::File>` in the current scope
   --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wasmtime-wasi-11.0.0/src/preview2/filesystem.rs:253:27
    |
253 |         let n = self.file.append_vectored(bufs)? as i64 as u64;
    |                           ^^^^^^^^^^^^^^^ method not found in `Arc<File>`

For more information about this error, try `rustc --explain E0599`.
```

I have no issues with version 9.x.x

Do you have an idea what the issue is?

**Comment by jornfranke** [comment]
Might be related to: https://github.com/bytecodealliance/wasmtime/issues/6370
?

**Comment by alexcrichton** [comment]
Thanks for the report! This is a mistake in our `Cargo.toml` where we have a listed version requirement on `system-interface = "0.25.0"` but the relevant functions were added in 0.25.2. This means that Wasmtime should actually depend on `system-interface = "0.25.2"`.

In the meantime though you should be able to fix this locally with `cargo update -p system-interface`. Does that fix the issue for you?

**Comment by jornfranke** [comment]
This works, thanks a lot. I can progress now with other applications - I use the sameple app as a minimal app to see impact of new wasmtime versions.

I am then also waiting for 11.0.1 or similar so I can avoid the additional cargo update step.

**Comment by alexcrichton** [comment]
This is fixed with 11.0.1
